### PR TITLE
Add pal_mount to System.Native

### PR DESCRIPTION
**Pull request by EgorBo**
Fixes https://github.com/mono/mono/issues/13030

### CODE REVIEW

**Comment by EgorBo** [comment]
Android:
```
 error: implicit declaration of function 'setmntent' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    FILE* fp = setmntent("/proc/mounts", MNTOPT_RO);
```

File: mono/native/pal_mount_tmp.c [hunk 1]
```diff
@@ -64,6 +64,9 @@ static int32_t GetMountInfo(MountPointFound onFound)
     }
 
     return result;
+#else
+    return 0;
```
Comment (akoeplinger) [inline mono/native/pal_mount_tmp.c]: does this mean this API will return empty result on Android?

**Comment by akoeplinger** [comment]
Please add tests for non-empty results for `Environment.GetDrives()` and `Directory.GetLogicalDrives()`.

`DriveInfo.GetDrives()` seems to be tested already.

File: mono/native/pal_mount_tmp.c (hunk 1 above)
Comment (EgorBo) [reply mono/native/pal_mount_tmp.c]: @akoeplinger as far as I understand, `setmntinfo` is not available in Android NDK

Comment (EgorBo) [reply mono/native/pal_mount_tmp.c]: It was added in Android-20 api level so I guess the `configure` check should handle it?..

Comment (akoeplinger) [reply mono/native/pal_mount_tmp.c]: Our previous code did somehow handle it though so we need to add that back to corefx and not regress functionality.

Here's the output I get from `Directory.GetLogicalDrives()` on XA 9.2:

```
/system
/vendor
/mnt/runtime/default/emulated
/mnt/runtime/read/emulated
/mnt/runtime/write/emulated
/cache
/data
/data/var/run/netns
/storage/emulated
```

Comment (akoeplinger) [reply mono/native/pal_mount_tmp.c]: Oh and I think we should put this code into corefx's `pal_mount.c` file instead of copy-pasting it into mono.

Comment (EgorBo) [reply mono/native/pal_mount_tmp.c]: That's why I put "_tmp_" in its name ;-) 
I guess it's better to use existing icall then, it's Android and AIX friendly.

File: mcs/class/corlib/Test/System/EnvironmentTest.cs [hunk 2]
```diff
@@ -239,5 +239,11 @@ public void MachineNameIsNotFullyQualifiedDomainName ()
 			Assert.AreEqual (-1, Environment.MachineName.IndexOf("."));
 		}
 #endif
+
+		[Test] // https://github.com/mono/mono/issues/13030
+		public void GetLogicalDrivesNotEmpty ()
+		{
+			Assert.AreNotEqual (0, Directory.GetLogicalDrives ().Length);
```
Comment (akoeplinger) [inline mcs/class/corlib/Test/System/EnvironmentTest.cs]: should be `Environment.GetLogicalDrives ()` here.

better would be to use `CollectionAssert.IsNotEmpty ()` so we get better message in case of failure.

Comment (EgorBo) [reply mcs/class/corlib/Test/System/EnvironmentTest.cs]: Oh, forgot about CollectionAssert thanks

File: mcs/class/corlib/Test/System/EnvironmentTest.cs [hunk 3]
```diff
@@ -239,5 +239,11 @@ public void MachineNameIsNotFullyQualifiedDomainName ()
 			Assert.AreEqual (-1, Environment.MachineName.IndexOf("."));
 		}
 #endif
+
+		[Test] // https://github.com/mono/mono/issues/13030
+		public void GetLogicalDrivesNotEmpty ()
+		{
+			CollectionAssert.IsNotEmpty (Directory.GetLogicalDrives ());
```
Comment (akoeplinger) [inline mcs/class/corlib/Test/System/EnvironmentTest.cs]: this still needs to be `**Environment**.GetLogicalDrives()`

**Comment by marek-safar** [comment]
@monojenkins backport 2019-02

**Comment by akoeplinger** [comment]
@monojenkins commit csproj
